src/parser/mod.rs

```diff
@@ -8530,6 +8535,9 @@ impl<'a> Parser<'a> {
                 with_privileges_keyword: self.parse_keyword(Keyword::PRIVILEGES),
             }
         } else {
+            let old_value = self.options.trailing_commas;
```

Comment:
I think this is a change we should discuss in a separate PR as I think it will cause some statements that used to parse to stop working (even if though it is more consistent)

Reply:
Although this was the original target for the PR, I lost track of the issue and ended up trying to address multiple points at once. I'm fine with delegating this to a separate PR.

Reply:
I don't see how that will break any existing statements. The original behavior was that if trailing commas were enabled and a DCL used 'SELECT', the parsing would fail.

This change temporarily ignores the trailing commas option when parsing DCL.

This is definitely not an ideal solution and should have its own PR (and therefore the corresponding issue should not be closed, even tho this was supposed to be the main target issue), but this has to be pushed with the new dialect trait method to pass the tests.